src/emmy/calculus/derivative.cljc

```diff
@@ -200,25 +200,34 @@
 ;; preserved through tag replacement and extraction.
 
 (extend-protocol d/IPerturbed
-  #?(:clj Fn :cljs function)
-  (perturbed? [#?(:cljs _ :clj f)]
-    #?(:clj (:perturbed? (meta f) false)
-       :cljs false))
+
+  ;; In Clojure, metadata can live directly on function objects.
+  Fn
+  (perturbed? [f] (:perturbed? (meta f) false))
+  (replace-tag [f old new] (replace-tag-fn f old new))
+  (extract-tangent [f tag] (extract-tangent-fn f tag))
+  MultiFn
+  (perturbed? [_] false)
   (replace-tag [f old new] (replace-tag-fn f old new))
   (extract-tangent [f tag] (extract-tangent-fn f tag))
 
   #?@(:cljs
-      [MetaFn
+      [;; In Clojurescript, we arrange for metadata to live directly on
+       ;; function objects by setting a special property and implementing
+       ;; IMeta and IWithMeta: see [[emmy.value]].
+       function
```

Comment:
can we still implement `perturbed?` here?

Reply:
yes (not sure how I overlooked that!)